pkg/controller/worker/machines.go

```diff
@@ -204,7 +206,7 @@ func (w *workerDelegate) generateMachineConfig() error {
 					}
 					// TODO: remove the csi topology label when AWS CSI driver stops using the aws csi topology key - https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/899
 					// add aws csi driver topology label if its not specified
-					return utils.MergeStringMaps(pool.Labels, map[string]string{awsCSIDriverTopologyKey: zone})
+					return utils.MergeStringMaps(pool.Labels, map[string]string{awsCSIDriverTopologyKey: zone, "networking.gardener.cloud/node-local-dns-enabled": nodeLocalDNSLabel})
```

Comment:
If you unconditionally include it in the map then the annotation will be there (even if the value in the kv will be ""). Is that the intended behavior ?